src/typescript-worker/util.ts

```diff
@@ -142,3 +142,24 @@ export const changeFileExtension = (path: string, newExt: string): string => {
   }
   return path.slice(0, -oldExt.length) + newExt;
 };
+
+/**
+ * Given a string and string-relative character index, return the equivalent
+ * line number and line-relative character index.
+ */
+export const charToLineAndChar = (
```

Comment:
Would there ever be a case where the newline is`\r\n` and does it matter?

Reply:
Good point, it could probably happen, but I think this implementation should still work fine*, since we'd consider the `\r` to be a regular character, and the `\n` would still determine the line boundaries. Added some test cases.

*Unless we want to handle commodore 64? https://en.wikipedia.org/wiki/Newline#Representation